src/ol/format/featureformat.js

```diff
@@ -165,21 +169,47 @@ ol.format.Feature.transformWithOptions = function(
       ol.proj.get(opt_options.featureProjection) : null;
   var dataProjection = opt_options ?
       ol.proj.get(opt_options.dataProjection) : null;
+  /**
+   * @type {ol.geom.Geometry|ol.Extent}
+   */
+  var transformed;
   if (featureProjection && dataProjection &&
       !ol.proj.equivalent(featureProjection, dataProjection)) {
     if (geometry instanceof ol.geom.Geometry) {
-      return (write ? geometry.clone() : geometry).transform(
+      transformed = (write ? geometry.clone() : geometry).transform(
           write ? featureProjection : dataProjection,
           write ? dataProjection : featureProjection);
     } else {
       // FIXME this is necessary because ol.format.GML treats extents
       // as geometries
-      return ol.proj.transformExtent(
+      transformed = ol.proj.transformExtent(
           write ? geometry.slice() : geometry,
           write ? featureProjection : dataProjection,
           write ? dataProjection : featureProjection);
     }
   } else {
-    return geometry;
+    transformed = geometry;
+  }
+  if (write && opt_options && opt_options.decimals) {
+    var power = Math.pow(10, opt_options.decimals);
+    // if decimals option on write, round each coordinate appropriately
+    if (Array.isArray(transformed)) {
```

Comment:
yes, that would be simpler, but I'm having trouble with the typedefs. If I just use your code, I get
```
actual parameter 1 of transformed.applyTransform does not match formal parameter
ERR! compile found   : function (?): undefined
ERR! compile required: function ((Array<number>|null), (Array<number>|null)=, number=): (Array<number>|null)
```
Presumably I need to add a `@param` to `transform()` but can't work out what form that should take to match the `TransformFunction` type.